### ISSUE TITLE
[Suggestion] Guidance on Container breakout vulnerabilities

### ISSUE THREAD

**Issue by raesene** [cncf/tag-security#881]
Description: As there have beena set of container breakout vulnerbiities in 2022, in a variety of parts of the stack, it's attracting more attention to this part of cloud native security. It could be good if the CNCF can provide some awareness about the risks of container breakout and possible mitigations that users of containerization technology can use to reduce the risks of this happening.

The topics for guidance might include :-

- Use of hardening techniques on containers (seccomp, Apparmor, SELinux)
- risks of user namespaces (some of the Linux CVEs could only be triggered from unprivileges users if user namespaces are available, which they are by default on popular distributions)
- Hardening by restricting access to features (the CRI-O bug relied on access to setting custom sysctls, which could be blocked by admission control)
- Importance of trusted images (the Containerd bug relies on the use of a crafted container image)
- Use of sandbox/virtualization to reduce risk (gVisor, Firecracker, Katacontainers) and where that might not be effective (e.g. the Containerd issue)

Impact: Help uses of containerization understand and mitigate the risks of container breakout

Scope: Not sure in detail but in principal perphaps a short white-paper/blog post on mitigations and risks, and then something for on-going awareness.

Additional info:
These blogs discuss the vulns and have links to the original disclosure and some ideas for mitigation

- [ContainerD Breakout](https://blog.aquasec.com/cve-2022-23648-containerd-cri-plugin)
- [CRI-O Breakout](https://blog.aquasec.com/cve-2022-0811-cri-o-vulnerability)
- [Linux Kernel - cgroups](https://blog.aquasec.com/new-linux-kernel-vulnerability-escaping-containers-by-abusing-cgroups)
- [Linux Kernel - "Dirty Pipe"](https://blog.aquasec.com/cve-2022-0847-dirty-pipe-linux-vulnerability) and more details and full PoC [from Datadog](https://www.datadoghq.com/blog/engineering/dirty-pipe-container-escape-poc/)
- [Linux Kernel - legacy_parse_param](https://blog.aquasec.com/cve-2022-0185-linux-kernel-container-escape-in-kubernetes)

STAG REP : @ashutosh-narkar

**Comment by lumjjb** [comment]
Tagging other folks that may be interested to chime in! @lizrice @sublimino @mhausenblas @antitree
@raravena80

**Comment by lumjjb** [comment]
Another thought that was brought up during the discussion was bug bounties as an incentive for these container escapes which may or may not be kernel bugs.

**Comment by sublimino** [comment]
This is a great idea, and touches on the difficulty of workload-specific container controls that is worthy of deeper discussion.

**Comment by lumjjb** [comment]
Let's develop a scope and then I think this is a good item to work on while it's topical, has attention of folks. 

I think a series of blog posts may be good as well! Maybe having a 2/3 part series:
- What are container breakouts, why should I care?
- What can i do to mitigate / protect against them
- (Stretch goal?) What are some future directions to make breakouts less impactful 

Tagging @ashutosh-narkar here who's started looking into a blog process for the TAG. Shall we use this as a pilot?

**Comment by lumjjb** [comment]
@ashutosh-narkar can you follow-up on getting this in progress

**Comment by raesene** [comment]
Another possibly useful resource on this one is  this [list of container breakout vulns](https://www.container-security.site/attackers/container_breakout_vulnerabilities.html)

**Comment by raesene** [comment]
I've started a Hackmd [note](https://hackmd.io/viQwq28URseBDb7b0YC6Rw) that I thought could be a good venue to gather ideas on this one.

In terms of content style any thoughts on what would work? I don't *think* this warrants an entire whitepaper style approach, perhaps a long blog, or whatever we'd call a short whitepaper?

**Comment by lumjjb** [comment]
I think a long blog would be appropriate. Ash is currently working on how
we can get these blogs up and the best way to publish them.

My thoughts is that there can be a long blog, attribution can be direct to
authors, or if the authors don't want to have direct attribution (for
approvals reasons) we can proxy the tag authorship. I guess we could call
it something else as well like technical doc and have an accompanying
announcement post

**Comment by raesene** [comment]
I'm still happy to work on this one once we've got a suitable venue/format :)

**Comment by raesene** [comment]
Hi @ashutosh-narkar sure. For the CNCF blog, do you know if they have any writing style guides that we should be following when drafting things up?

**Comment by lumjjb** [comment]
AFAIK it's only style guidelines around logo and certain cloud native
terminology and code of conduct guidelines

Otherwise I believe it's pretty flexible.

**Comment by fidencio** [comment]
Hey!  I'm from the Kata Containers side and I'd be interested to help on whatever may be needed.

Noted, I've been working on Kata Containers + SELinux support (which touched CRI-O and containerd) and I'm very much interested in the outcome of these articles.

**Comment by ashutosh-narkar** [comment]
Hello @raesene do you have any update on this? Please let us know if you need any help. Thanks!

**Comment by raesene** [comment]
hi, no update at this precise moment in time, I've hit a busy patch so have less availability for this, I do plan to return to it though.

**Comment by ashutosh-narkar** [comment]
Sounds good. Thanks @raesene.

**Comment by mnm678** [comment]
Closing due to inactivity. Please re-open if there is renewed interest.
